packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -1067,25 +1133,25 @@ export class GarbageCollector implements IGarbageCollector {
 			}
 		}
 
-		if (!this.configs.shouldRunSweep) {
-			return sweepPhaseStats;
-		}
-
 		// If sweep is enabled, the counts from the mark phase stats do not include nodes that have been
-		// deleted in previous runs. Add the deleted node counts to life time stats.
+		// deleted in previous runs. So, add the deleted node counts to life time stats.
 		sweepPhaseStats.lifetimeNodeCount += sweepPhaseStats.deletedNodeCount;
 		sweepPhaseStats.lifetimeDataStoreCount += sweepPhaseStats.deletedDataStoreCount;
 		sweepPhaseStats.lifetimeAttachmentBlobCount += sweepPhaseStats.deletedAttachmentBlobCount;
 
-		// The node deleted in current run are counted twice - once in allDeletedNodes and again in
-		// markPhaseStats. So, remove them from the life time stats.
+		if (this.configs.sweepEnabled) {
+			return sweepPhaseStats;
+		}
+
+		// If sweep is not enabled, the currently deleted node stats should be added to deleted stats.
```

Comment:
You sure we want to do that?  I would think just let these stats tell the official/synchronized story.  As it is, it's kind of optimistically assuming the GC op will make it.